internal/rtree/index.go

```diff
@@ -209,6 +184,122 @@ func (i *Index) Rebuild() error {
 	return nil
 }
 
+func handleUpdateRemotes(repo Repo) (Repo, error) {
+	currentRemotes, err := collectRemotesFromAbsolutePath(repo.AbsolutePath())
+	if err != nil {
+		return Repo{}, err
+	}
+
+	var (
+		newRemotes []Remote
+		toUpdate   []string
+	)
+
+configRemotes:
+	for _, remote := range repo.Remotes {
+		// remote in index exists in repo
+		for _, currentRemote := range currentRemotes {
+			if currentRemote.URL.CompactURL() == remote.URL.CompactURL() {
+				// add URL matching remote as the name might have changed
+				// e.g. when forking from upstream and renaming the old origin to upstream
+				newRemotes = append(newRemotes, currentRemote)
+				continue configRemotes
+			}
+		}
+
+		// ask the user what to do with remotes not existing in checked out repo
+		selection, err := cli.Interface.Query(
+			fmt.Sprintf(`repository "%s" remote "%s" has been deleted`, repo.AbsolutePath(), remote.Name),
```

Comment:
```suggestion
			fmt.Sprintf(`repository %q remote %q has been deleted`, repo.AbsolutePath(), remote.Name),
```

Also, I suggest the clearer phrasing `remote %q of repository %q has been deleted`.